src/main/java/com/gamegoo/service/manner/MannerService.java

```diff
@@ -0,0 +1,159 @@
+package com.gamegoo.service.manner;
+
+import com.gamegoo.apiPayload.code.status.ErrorStatus;
+import com.gamegoo.apiPayload.exception.handler.MannerHandler;
+import com.gamegoo.apiPayload.exception.handler.MemberHandler;
+import com.gamegoo.apiPayload.exception.handler.TempHandler;
+import com.gamegoo.domain.Member;
+import com.gamegoo.domain.manner.MannerKeyword;
+import com.gamegoo.domain.manner.MannerRating;
+import com.gamegoo.domain.manner.MannerRatingKeyword;
+import com.gamegoo.dto.manner.MannerRequest;
+import com.gamegoo.repository.manner.MannerKeywordRepository;
+import com.gamegoo.repository.manner.MannerRatingKeywordRepository;
+import com.gamegoo.repository.manner.MannerRatingRepository;
+import com.gamegoo.repository.member.MemberRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class MannerService {
+    private final MemberRepository memberRepository;
+    private final MannerRatingRepository mannerRatingRepository;
+    private final MannerRatingKeywordRepository mannerRatingKeywordRepository;
+    private final MannerKeywordRepository mannerKeywordRepository;
+
+    public MannerRating insertManner(MannerRequest.mannerInsertDTO request, Long memberId){
+
+        Member member = memberRepository.findById(memberId).orElseThrow(()->new MemberHandler(ErrorStatus.MEMBER_NOT_FOUND));
+
+        // 매너평가를 받는 회원 존재 여부 검증.
+        Member targetMember = memberRepository.findById(request.getToMemberId()).orElseThrow(()->new MemberHandler(ErrorStatus.MANNER_TARGET_MEMBER_NOT_FOUND));
+
+        // 매너평가를 받는 회원 탈퇴 여부 검증.
+        if (targetMember.getBlind()){
+            throw new MemberHandler(ErrorStatus.USER_DEACTIVATED);
+        }
+
+        // 매너평가 최초 시도 여부 검증.
+        List<MannerRating> mannerRatings = mannerRatingRepository.findByToMemberId(targetMember.getId());
+        if (!mannerRatings.isEmpty()){
+            for (MannerRating mannerRating : mannerRatings) {
+                if (mannerRating.getFromMember().getId().equals(member.getId())){
+                    List<MannerRatingKeyword> mannerKeywords = mannerRatingKeywordRepository.findByMannerRating(mannerRating);
+                    if (!mannerKeywords.isEmpty()){
+                        MannerRatingKeyword mannerKeyword = mannerKeywords.get(0);  // 첫번째 키워드만 조회.
+                        if (mannerKeyword.getMannerKeyword().getIsPositive()){
+                            throw new MannerHandler(ErrorStatus.MANNER_CONFLICT);
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
이부분을 그냥 MannerRating 엔티티에 isPositive 칼럼을 추가하고, 
```suggestion
List<MannerRating> mannerRatings = mannerRatingRepository.findByFromMemberIdAndToMemberId(member.getId(), targetMember.getId());
        List<MannerRating> positiveMannerRatings = mannerRatings.stream()
            .filter(MannerRating::isPositive)
            .collect(Collectors.toList());
        
        if(!positiveMannerRatings.isEmpty()){
            throw new MannerHandler(ErrorStatus.MANNER_CONFLICT);
        }
```

이런식으로 변경하는 건 어떨까요..?! 이 코드가 정상 작동할지 테스트는 못해봤는데 대충 이런 방식도 가능할 것 같아서 남깁니다..!

Reply:
현재 ERD - 매너 키워드에  isPositive 컬럼이 존재하는데, MannerRating에도 추가를 했으면 좋겠다는 말씀이실까요?

Reply:
[코드 리팩토링 완료]